internal/sim/sim.go

```diff
@@ -89,14 +90,16 @@ type Simulator struct {
 	ctx   context.Context // simulation context
 	group *errgroup.Group // group with all running goroutines
 
-	mu          sync.Mutex // guards the following fields
-	rand        *rand.Rand // random number generator
-	numOps      int        // number of spawned ops
-	calls       []*call    // pending calls
-	replies     []*reply   // pending replies
-	history     []Event    // history of events
-	nextTraceID int        // next trace id
-	nextSpanID  int        // next span id
+	mu          sync.Mutex       // guards the following fields
+	rand        *rand.Rand       // random number generator
+	current     int              // currently running op
+	numStarted  int              // number of started ops
+	notFinished map[int]struct{} // set of not finished ops
```

Comment:
Can you explain what these are keyed by and why they are a map and not a slice? (To be able to detect quickly when everything is finished.)